src/NzbDrone.Core.Test/DecisionEngineTests/UpgradeAllowedSpecificationFixture.cs

```diff
@@ -0,0 +1,293 @@
+        [Test]
+        public void should_return_false_when_quality_are_the_same_language_is_better_and_upgrade_allowed_is_false_for_language_profile()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English, Language.French),
+                    Cutoff = Language.French,
+                    UpgradeAllowed = false
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.French
+            ).Should().BeFalse();
+        }
+
+        [Test]
+        public void should_return_false_when_quality_is_better_languages_are_the_same_and_upgrade_allowed_is_false_for_quality_profile()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = false
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English),
+                    Cutoff = Language.English,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.FLAC),
+                Language.English
+            ).Should().BeFalse();
+        }
+
+        [Test]
+        public void should_return_true_for_language_upgrade_when_upgrading_is_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English, Language.French),
+                    Cutoff = Language.French,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.French
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_same_language_when_upgrading_is_allowed()
```

Comment:
Should this be `false` if it's not a strict upgrade?

Reply:
Seems the idea behind all this was whether or not to allow the update if release is an update. The actual logic to determine if its upgradable is handled int he UpgradeDiskSpecification. So it gets caught there and rejected as being "not an upgrade to current file on disk".  Instead of being rejected in the "isUpdateAllowed" spec.